Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -902,13 +902,17 @@ void NetworkConnectionToWebProcess::domCookiesForHost(const URL& url, Completion
 
 #if HAVE(COOKIE_CHANGE_LISTENER_API)
 
-void NetworkConnectionToWebProcess::subscribeToCookieChangeNotifications(const String& host)
+void NetworkConnectionToWebProcess::subscribeToCookieChangeNotifications(const URL& url, const URL& firstParty, WebCore::FrameIdentifier frameID, WebCore::PageIdentifier pageID, WebCore::ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, CompletionHandler<void(bool)>&& completionHandler)
 {
-    ASSERT(!m_hostsWithCookieListeners.contains(host));
```

Comment:
As Sihui explained, we dropped this assertion because now the web process can send request for the same host, but with different url / firstParty / frameID / pageID / shouldRelaxThirdPartyCookieBlocking. 

This means that in `NetworkStorageSessionCocoa::startListeningForCookieChangeNotifications`, we may attempt to add this same observer again. prior this code change, we had an assert to ensure this never happened. Which was ok, because if the same host was encountered again, it would fail here in `NetworkConnectionToWebProcess::subscribeToCookieChangeNotifications`. Without this first assert, some tests fail since they try to subscribe the same observer with different url, etc).

But since we still want to try listening for cookies with the potential different url and so on, we no longer want to have the assert for the observer in `startListeningForCookieChangeNotifications`, so we fix the failures by removing the assert but still only adding the observer if it hasn't already been added.